sklearn/linear_model/_ridge.py

```diff
@@ -1129,6 +1129,12 @@ def __init__(self, alphas=(0.1, 1.0, 10.0), *,
         self.gcv_mode = gcv_mode
         self.store_cv_values = store_cv_values
         self.is_clf = is_clf
+        self.alpha_per_target = alpha_per_target
+
+        # alpha_per_target cannot be used in classifier mode. All subclasses
+        # of _RidgeGCV that are classifiers keep alpha_per_target at its
+        # default value: False, so the condition below should never happen.
+        assert not (is_clf and alpha_per_target)
```

Comment:
We normally do not validate in `init`. This can be moved to the `fit` method while raising an `ValueError`.